test/pipeline-utils/wrapTextNodes.test.html

```diff
@@ -0,0 +1,166 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+  <script src="/some/path/scripts/scripts.js"></script>
+</head>
+
+<body>
+  <main>
+    <div>
+      <div class="text original">
+        <div>
+          <div>This will get a wrapping p</div>
+          <div><p>This will get a not get wrapping p since it already has one</p></div>
+          <div>  </div> <!-- I'm not sure pipeline can produce this structure, but adding it JIC -->
+        </div>
+      </div>
+      <div class="text expected">
+        <div>
+          <div><p>This will get a wrapping p</p></div>
+          <div><p>This will get a not get wrapping p since it already has one</p></div>
+          <div><p>  </p></div> <!-- I'm not sure pipeline can produce this structure, but adding it JIC -->
+        </div>
+      </div>
+      <div class="icons original">
+        <div>
+          <div><span class="icon icon-name"></span> it also works when the icon comes first</div>
+          <div><span class="icon icon-name"></span></div>
+        </div>
+      </div>
+      <div class="icons expected">
+        <div>
+          <div><p><span class="icon icon-name"></span> it also works when the icon comes first</p></div>
+          <div><p><span class="icon icon-name"></span></p></div>
+        </div>
+      </div>
+      <div class="code original">
+        <div>
+          <div><code>code blocks also get wrapped</code></div>
+          <div>text with some <code>inline code block also get wrapped</code></div>
```

Comment:
one callout on `code` elements: In one revision of this, I was excluding `code` from being wrapped similar to `picture` but decided against it because of the possibility a text node with a `<code>` sibling as shown here. 

If we wanted to get surgical, we could probably only wrap the code elements when they do have a text sibling, but that would add some complexity to the logic that I'm not sure is necessary.